File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        if(head==null){
            return null;
        }
        ListNode reversed = new ListNode(head.val);
        ListNode og = head.next;
        
        while(og!=null){
            ListNode head2 = new ListNode(og.val,reversed);
            reversed=head2;
            og=og.next;
        }
        return reversed;
    }
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        length = 0
        curr = head
        while curr:
            length+=1
            curr = curr.next
        p = length-n
        curr = head
        if p==0:
            return head.next
        while p>0:
            prev = curr
            curr = curr.next
            p-=1
        if not curr:
            prev.next = None
        else:
            prev.next = curr.next
        
        return head

        